File: src/main/java/net/dontdrinkandroot/gitki/wicket/component/FileActionsDropDownButton.java
package net.dontdrinkandroot.gitki.wicket.component;

import net.dontdrinkandroot.gitki.model.FilePath;
import net.dontdrinkandroot.gitki.model.Role;
import net.dontdrinkandroot.gitki.wicket.component.item.EditFileItem;
import net.dontdrinkandroot.gitki.wicket.component.item.RemoveFileModalItem;
import net.dontdrinkandroot.gitki.wicket.security.Render;
import net.dontdrinkandroot.wicket.bootstrap.component.button.DropDownButton;
import net.dontdrinkandroot.wicket.bootstrap.css.DropDownAlignment;
import org.apache.wicket.markup.repeater.RepeatingView;
import org.apache.wicket.model.IModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Render(Role.COMMITTER)
public class FileActionsDropDownButton extends DropDownButton<FilePath>
{
    public FileActionsDropDownButton(String id, IModel<FilePath> model)
    {
        super(id, model, null);
        this.setDropDownAlignment(DropDownAlignment.RIGHT);
    }

    @Override
    protected void populateItems(RepeatingView itemView)
    {
        itemView.add(new EditFileItem(itemView.newChildId(), this.getModel()));
        itemView.add(new RemoveFileModalItem(itemView.newChildId(), this.getModel()));
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/component/FileActionsDropdownButton.kt
package net.dontdrinkandroot.gitki.wicket.component

import net.dontdrinkandroot.gitki.model.FilePath
import net.dontdrinkandroot.gitki.model.Role
import net.dontdrinkandroot.gitki.wicket.component.item.EditFileItem
import net.dontdrinkandroot.gitki.wicket.component.item.MoveFileModalItem
import net.dontdrinkandroot.gitki.wicket.component.item.RemoveFileModalItem
import net.dontdrinkandroot.gitki.wicket.security.Enable
import net.dontdrinkandroot.wicket.bootstrap.component.button.DropdownButton
import net.dontdrinkandroot.wicket.bootstrap.css.ButtonSize
import net.dontdrinkandroot.wicket.bootstrap.css.ButtonStyle
import net.dontdrinkandroot.wicket.bootstrap.css.DropdownAlignment
import net.dontdrinkandroot.wicket.css.CssClass
import net.dontdrinkandroot.wicket.kmodel.model
import org.apache.wicket.markup.repeater.RepeatingView
import org.apache.wicket.model.IModel
import org.apache.wicket.model.Model
import net.dontdrinkandroot.wicket.bootstrap.css.FontAwesome5IconClass as Fa5IconClass

@Enable(Role.COMMITTER)
class FileActionsDropdownButton(
    id: String, model: IModel<FilePath>,
    buttonStyleModel: IModel<ButtonStyle> = Model(null),
    buttonSize: ButtonSize? = null,
    prependIconModel: IModel<CssClass> = model(Fa5IconClass.ELLIPSIS_V.createIcon(fixedWidth = true)),
    appendIconModel: IModel<CssClass> = Model(null),
    dropdownAlignment: DropdownAlignment? = null
) : DropdownButton<FilePath>(
    id,
    model,
    null,
    buttonStyleModel,
    Model(buttonSize),
    prependIconModel,
    appendIconModel,
    Model(dropdownAlignment)
) {

    override fun populateItems(itemView: RepeatingView) {
        itemView.add(EditFileItem(itemView.newChildId(), this.model))
        itemView.add(MoveFileModalItem(itemView.newChildId(), this.model))
        itemView.add(RemoveFileModalItem(itemView.newChildId(), this.model))
    }
}